code/modules/aspects/aspects.dm

```diff
@@ -0,0 +1,119 @@
+// Aspects are basically skills + stats + feats all rolled into one. You get to choose a certain number
+// of them at character generation and they will alter some interactions with the world. Very WIP.
+/decl/aspect
+	var/name = "Filler Aspect"                        // Name/unique index.
+	var/desc = "An aspect is a kind of skill."        // Flavour text.
+	var/use_icon_state = ""                           // State to use for the above.
+	var/icon/use_icon                                // I REALLY HATE \icon A LOT.
+	var/category                                      // Header for root aspects in char prefs.
+
+	var/parent_name
+	var/decl/aspect/parent
+	var/list/children = list()
+
+/decl/aspect/New()
+	..()
+	use_icon = icon(icon = 'icons/obj/aspects.dmi', icon_state = use_icon_state)
+
+/decl/aspect/Destroy()
+	children.Cut()
+	parent = null
+	return ..()
+
+/decl/aspect/proc/do_post_spawn(var/mob/living/carbon/human/holder)
+	return
+
+// Called by preferences selection for HTML display.
+/decl/aspect/proc/get_aspect_selection_data(var/caller, var/list/ticked_aspects = list(), var/recurse_level = 0, var/ignore_children_if_unticked = 1, var/ignore_unticked)
+
+	var/ticked = (name in ticked_aspects)
+	if(ignore_unticked)
+		return ""
+
+	var/result = "<tr><td width=50%>"
+	if(recurse_level)
+		for(var/x = 1 to recurse_level)
+			result += "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" // Padding.
+
+	if(caller)
+		result += "<img src='[name]_small.png'>&nbsp&nbsp<a href='?src=\ref[caller];toggle_aspect=[name]'><font color='[ticked ? "#E67300" : "#3366CC"]'>[name]</font></a>"
+	else
+		result += "<img src='[name]_small.png'>&nbsp&nbsp<font color='[ticked ? "#E67300" : "#3366CC"]'>[name]</font>"
+
+	result += "</td><td>"
+	if(ticked)
+		result += "<font size=1><i>[desc]</i></font>"
+	result += "</td></tr>"
+	if(children.len && !(ignore_children_if_unticked && !ticked))
+		for(var/decl/aspect/A in children)
+			result += A.get_aspect_selection_data(caller, ticked_aspects, (recurse_level+1), ignore_children_if_unticked)
+	return result
+
+// Helpers.
+/datum/mind/var/list/aspects = list()
+
+/mob/proc/has_aspect(var/skillname)
```

Comment:
It would require some rewrite in the load/save step but I wonder if it wouldn't be more robust to store the aspect types in the list instead of just their names.
That way one could do has_aspect(/decl/aspect/.../brawler), avoiding typos without also having too define new constants.